app/server/tests/api-getUser.test.js

```diff
@@ -0,0 +1,130 @@
+const { AuthFunctions } = require('./functions/authFunctions');
+const user = require('./functions/userFunctions');
+const { password, name, email } = require('./functions/randomizer');
+
+let auth = new AuthFunctions();
+
+jest.setTimeout(10000); // Some register-login-action groups take longer than 5000ms default.
+
+describe('Get the current user\'s information with /user', () => {
+    let userName = name.gen();
+    let userPassword = password.gen();
+    let loginResponse;
+    let response; 
+    beforeAll(async () => {
+        await auth.register(userName, email.gen(), userPassword);
+        loginResponse = await auth.login(userName, userPassword);
+        response = await user.getUser(loginResponse.body.token);
+    });
+
+    afterAll(async () => {
+        await auth.deleteUsers();
+    });
+
+    test('User is successfully found - returns 200', () => {
+        expect(response.status).toBe(200);
+    });
+
+    test('Body of response is formated as expected (returns only set keys)', () => {
+        expect(typeof response.body).toBe('object');
+        expect(response.body.name).toBeTruthy();
+        expect(response.body.email).toBeTruthy();
+        expect(response.body.first_name).not.toBeTruthy();
+        expect(response.body.last_name).not.toBeTruthy();
+        expect(response.body.bio).not.toBeTruthy();
+        expect(response.body.title).not.toBeTruthy();
+        expect(response.body.quote).not.toBeTruthy();
+    });
+
+    xtest('After setting all keys, they are returned by the /user endpoint', async () => {
+        let body = {
+            "email": email.gen(),
+            "first_name": "newFirstName",
+            "last_name": "newLastName",
+            "bio": "new bio",
+            "title": "new title"
+        };
+        let editResponse = await user.editUserByObject(loginResponse.body.token, body);
+        expect(editResponse.status).toBe(204);
+        response = await user.getUser(loginResponse.body.token);
+        expect(response.body.name).toBeTruthy();
+        expect(response.body.email).toBeTruthy();
+        expect(response.body.first_name).toBeTruthy();
+        expect(response.body.last_name).toBeTruthy();
+        expect(response.body.bio).toBeTruthy();
+        expect(response.body.title).toBeTruthy();
+        // expect(response.body.quote).toBeTruthy(); // currently missing from User schema
```

Comment:
TODO

Reply:
I am assuming its the developers that are going to be developing the scheme change and you that turns on the test when it's available.